src/math/transform.cpp

```
@@ -0,0 +1,164 @@
+    }
+    if (scaleY > 0.0f) {
+        rotMat.at(0, 1) /= scaleY;
+        rotMat.at(1, 1) /= scaleY;
+        rotMat.at(2, 1) /= scaleY;
+    }
+    if (scaleZ > 0.0f) {
+        rotMat.at(0, 2) /= scaleZ;
+        rotMat.at(1, 2) /= scaleZ;
+        rotMat.at(2, 2) /= scaleZ;
+    }
+
```

Cppcheck (2.19.0)

[warning] The address of variable 'x' might be accessed at non-zero index.

(objectIndex)